src/Microsoft.ML.Data/Transforms/ColumnSelecting.cs

```diff
@@ -682,8 +677,13 @@ public RowCursor[] GetRowCursorSet(IEnumerable<Schema.Column> columnsNeeded, int
 
             void ICanSaveModel.Save(ModelSaveContext ctx) => _transform.SaveModel(ctx);
 
-            public Func<int, bool> GetDependencies(Func<int, bool> activeOutput)
+            /// <summary>
+            /// Given a set of columns, return the input columns that are needed to generate those output columns.
+            /// </summary>
+            IEnumerable<Schema.Column> IRowToRowMapper.GetDependencies(IEnumerable<Schema.Column> columns)
             {
+                var activeOutput = RowCursorUtils.FromColumnsToPredicate(columns, _mapper.OutputSchema);
```

Comment:
>FromColumnsToPredicate [](start = 50, length = 22)

You will need to untangle it sooner or later :)
Right now I don't see any reason to use predicate. We get set of columns we return set of columns, and we don't call any function which required predicate.
But you can always postpone it to moment when we delete than Utils method

Reply:
You know, that Utils method might stay for a while, because GetActive is not public... 

untangling this now:)

---
In reply to: [256133342](https://github.com/dotnet/machinelearning/pull/2504#discussion_r256133342) [](ancestors = 256133342)